lib/src/configuration.dart

```diff
@@ -429,18 +429,22 @@ extension SchemaObjectInternal on SchemaObject {
 /// This can happen if the server is rolled back or restored from backup.
 /// {@category Sync}
 abstract class ClientResetHandler {
-  /// Defines what should happen in case of a Client reset
-  ClientResyncModeInternal get _mode;
+  
+  // Defines what should happen in case of a Client reset
+  final ClientResyncModeInternal _mode;
 
-  BeforeResetCallback? get _onBeforeReset => null;
-  AfterResetCallback? get _onAfterDiscard => null;
-  AfterResetCallback? get _onAfterRecovery => null;
+  final BeforeResetCallback? _onBeforeReset;
+  final AfterResetCallback? _onAfterDiscard;
+  final AfterResetCallback? _onAfterRecovery;
```

Comment:
I think I prefer this as well, although it can be argued it allocates a bit more memory

Reply:
yes, but since there will be a limited number of instances of these classes throughout the app lifetime its should be negligible.